### PR TITLE
[DISLIBSVM] Add Distributed Read/Write LIBSVM on Spark

### PR DESCRIPTION
**Pull request by fathollahzadeh**
1. Allows mapping RDD partitions of LIBSVM rows into a set of partial binary blocks
2. Add file format properties to support custom delimiters between data instances and index value
3. Update Read/Write CP instructions to support the file format properties

### CODE REVIEW

**Comment by phaniarnab** [comment]
Looks like all tabs are replaced by spaces in all the files your commits touched. Your IDE probably has automatically applied these changes. Can you please have a look @fathollahzadeh? In addition to that, indentation is broken in the new classes.

**Comment by fathollahzadeh** [comment]
> Looks like all tabs are replaced by spaces in all the files your commits touched. Your IDE probably has automatically applied these changes. Can you please have a look @fathollahzadeh? In addition to that, indentation is broken in the new classes.

@phaniarnab Thanks for reviewing the code. I'll fix the changes.

**Comment by phaniarnab** [comment]
Thanks, @fathollahzadeh for fixing the space issues. I had a look into the changes and it already looks quite good. However, it would be great if you can fix the automatic application of code style to all the files. Currently, it is very hard to spot the actual changes done by your commits. In addition to that, the code style introduced some weird formatting all over the common files. Other than that, the file VariableCPInstruction.java still has spaces instead of tabs.
Meanwhile, I will try to have a detailed look into the changes, run some tests, rebase, resolve the conflicts and make a commit to run the tests in github.

File: src/test/java/org/apache/sysds/test/functions/io/libsvm/WriteLIBSVMTest.java [hunk 1]
```diff
@@ -27,70 +27,70 @@
...
-      runTest(true, false, null, -1);
-    }
-    finally {
-      rtplatform = oldPlatform;
-      CompilerConfig.FLAG_PARREADWRITE_TEXT = oldpar;
-      DMLScript.USE_LOCAL_SPARK_CONFIG = sparkConfigOld;
-    }
-  }
+			runTest(true, false, null, -1);
+		}
```
Comment (phaniarnab) [inline src/test/java/org/apache/sysds/test/functions/io/libsvm/WriteLIBSVMTest.java:89]: Can you please add a way to verify if the written data is correct? You can read the output file in a R script and compare the sums. You can follow the example of WriteCSVTest.

**Comment by fathollahzadeh** [comment]
> Thanks, @fathollahzadeh for fixing the space issues. I had a look into the changes and it already looks quite good. However, it would be great if you can fix the automatic application of code style to all the files. Currently, it is very hard to spot the actual changes done by your commits. In addition to that, the code style introduced some weird formatting all over the common files. Other than that, the file VariableCPInstruction.java still has spaces instead of tabs.
> Meanwhile, I will try to have a detailed look into the changes, run some tests, rebase, resolve the conflicts and make a commit to run the tests in github.

Thanks, @phaniarnab  for the review. I remove all unnecessary code styles in changed files. Now, the code style applied just for the task updates.

File: src/test/java/org/apache/sysds/test/functions/io/libsvm/WriteLIBSVMTest.java (hunk 1 above)
Comment (fathollahzadeh) [reply src/test/java/org/apache/sysds/test/functions/io/libsvm/WriteLIBSVMTest.java]: Tests added to verify read/write. Thank!.

**Comment by phaniarnab** [comment]
LGTM. Thanks, @fathollahzadeh for the patch. While merging I fixed a few formating issues.

**Pull request closed without merging**